### ISSUE TITLE
register*StateChangeHandler doesn't work on Android since cordova-android@6.3.0 

### ISSUE THREAD

**Issue by iwan-uschka** [dpa99c/cordova-diagnostic-plugin#279]
**I'm submitting a ...**  (check one with "x"):
- [x] bug report
- [ ] feature request
- [ ] documentation issue

# Bug report

**Current behavior:**

- build android project with cordova-android@6.3.0 (Android target: android-26) => registerLocationStateChangeHandler() does not work, neither does registerBluetoothStateChangeHandler()

**Expected behavior:**
- registerLocationStateChangeHandler() should work like within cordova-android@6.2.3 build (Android target: android-25)

**Steps to reproduce:**

**Environment information**

- Cordova CLI version 
	- `cordova -v`=> 7.1.0
- Cordova platform version
	- `cordova platform ls` => android 6.3.0
- Plugins & versions installed in project (including this plugin)
    - `cordova plugin ls`
cordova-android-support-gradle-release 1.2.0 "cordova-android-support-gradle-release"
cordova-plugin-device 2.0.1 "Device"
cordova-plugin-device-orientation 2.0.0 "Device Orientation"
cordova-plugin-facebook4 1.7.4 "Facebook Connect"
cordova-plugin-inappbrowser 2.0.2 "InAppBrowser"
cordova-plugin-network-information 2.0.1 "Network Information"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.1 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova.plugins.diagnostic 3.9.2 "Diagnostic"
ionic-plugin-keyboard 2.2.1 "Keyboard"
- Dev machine OS and version, e.g.
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G19009
        
_Runtime issue_
- Device details
    - LG Nexus 5X
- OS details
    - Android 8.1
	
_Android build issue:_	
- Node JS version
    - `node -v` => v9.4.0
- Gradle version
	- `ls platforms/android/.gradle` => 2.14  3.3
- Target Android SDK version
	- `android:targetSdkVersion` in `AndroidManifest.xml` => 25

**Comment by iwan-uschka** [comment]
Just checked cordova-diagnostic-plugin@4 and the error still occures.

**Comment by dpa99c** [comment]
Seems that the Android APIs have changed since API 26, as stated [here](https://stackoverflow.com/a/48749980/777265):

> there's no way on Android API 26 to listen to implicit broadcasts like `<action android:name="android.location.PROVIDERS_CHANGED" />` if the app is killed or isn't currently running in background.

Since the plugin currently uses [implicit broadcast filters](https://github.com/dpa99c/cordova-diagnostic-plugin/blob/master/plugin.xml#L293) I'll need to rewrite it to register broadcast receivers at runtime. Thanks Google...

**Comment by iwan-uschka** [comment]
Thanks for taking care of this one. And yeah, thanks google...

**Comment by dpa99c** [comment]
Only now did I spot the [Broadcast Limitations](https://developer.android.com/about/versions/oreo/background.html#broadcasts) section in the Android 8.0 documentation.

Anyway, I'm working on this now so hope to push out a patch version some time today.

**Comment by dpa99c** [comment]
Fix published in v4.0.1

**Comment by iwan-uschka** [comment]
Ahh, i wasn't able to test it yet but i want to say thank you!

**Comment by EdwinRozario** [comment]
@dpa99c I have noticed an odd behaviour `registerLocationStateChangeHandler()` for Android across all devices in `v5.0.0`. We are using Ionic4 with `cordova-android@8.0.0`.

![permission_modal](https://user-images.githubusercontent.com/264455/66955874-c2417400-f0bf-11e9-8a59-0a937c44f3fe.png)

1. The above page requires location permission and the modal is presented to the user. The app is listening to `registerLocationStateChangeHandler()` so that a map can be rendered in this page.

![nearme_page](https://user-images.githubusercontent.com/264455/66955963-f1f07c00-f0bf-11e9-8308-2884004e46c8.png)

2. After allowing permission `registerLocationStateChangeHandler()` dosent seem to have fired so we still have the open setting button.

![settings](https://user-images.githubusercontent.com/264455/66956046-25330b00-f0c0-11e9-9c56-6dafae351036.png)

3. Opening Setting and allowing location.

![relaunch](https://user-images.githubusercontent.com/264455/66956075-3c71f880-f0c0-11e9-8b03-23c91be7baea.png)

4. Relaunching the application after the settings change works fine.

I tried to debug this. It seems `registerLocationStateChangeHandler()` dosent receive any state change when allowing permission from the modal popup but it receives state change when allowing it from settings.

Sorry if this is the wrong place to mention this issue. It looked like it had to do with `registerLocationStateChangeHandler()`.
